File: SolarCoffee.Data/Models/Customer.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace SolarCoffee.Data.Models
{
    public class Customer
    {
        public int Id { get; set; }
        [MaxLength(100)]
        public string FirstName { get; set; }
        [MaxLength(100)]
        public string LastName { get; set; }
        public CustomerAddress PrimaryAddress { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime UpdatedOn { get; set; }
    }
}

File: solarcoffee.frontend/src/models/request/customer/customer.ts
import ICustomerAddress from "./customer-address";

export default interface ICustomer {
	id: number;

	primaryAddress: ICustomerAddress;

	firstName: string;
	lastName: string;
	email: string;
	phoneNumber: string;

	createdOn: Date;
	updatedOn: Date;
}
